src/TeamSpeak3QueryApi/Query/Protocols/IProtocol.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace TeamSpeak3QueryApi.Net.Query.Protocols
+{
+    public interface IProtocol
+    {
+        Task<CancellationTokenSource> ConnectAsync();
```

Comment:
That interface doesn't make any sense IMHO when only either of these Methods can be implemented.